skfp/fingerprints/e_state.py

```diff
@@ -0,0 +1,54 @@
+from typing import Union
+
+import numpy as np
+import pandas as pd
+import scipy.sparse as spsparse
+
+from skfp.fingerprints.base import FingerprintTransformer
+
+
+class EStateFingerprint(FingerprintTransformer):
+    """this fingerprint returns 2 arrays:
+    The first (of ints) contains the number of times each possible atom type is hit
+     The second (of floats) contains the sum of the EState indices for atoms of
+    """
+
+    def __init__(
+        self,
+        variant: str = "sum",
+        sparse: bool = False,
+        n_jobs: int = None,
+        verbose: int = 0,
+        random_state: int = 0,
+        count: bool = False,
```

Comment:
How does the combination of `variant` and `count` work? I think we should raise an error for `variant="sum"` with `count=True`, since this combination doesn't make sense.

Reply:
That's the problematic part. We have the `count` member of the parent class and `count` option for `variant` and that introduces a redundancy problem. I think we could have 2 possible solutions to that. One would be removing the original `count` from the constructor and initializing it as false in the parent class constructor call. The other option would be to ignore the `variant` when `count` is true and remove it as a legal variant, leaving only `"sum"` and `"binary"`

Reply:
Nevermind. I read comments below and the `count` set to `true` will be legal with `"binary"` variant